device-types/Cisco/WS-C3750E-24PD.yaml

```diff
@@ -0,0 +1,121 @@
+---
+manufacturer: Cisco
+model: Catalyst 3750E-24PD
+slug: ws-c3750e-24pd
+part_number: WS-C3750E-24PD
+is_full_depth: false
+u_height: 1
+interfaces:
+  - name: GigabitEthernet1/0/1
+    type: 1000base-t
+    poe_mode: pse
+    poe_type: type1-ieee802.3af
+  - name: GigabitEthernet1/0/2
+    type: 1000base-t
+    poe_mode: pse
+    poe_type: type1-ieee802.3af
+  - name: GigabitEthernet1/0/3
+    type: 1000base-t
+    poe_mode: pse
+    poe_type: type1-ieee802.3af
+  - name: GigabitEthernet1/0/4
+    type: 1000base-t
+    poe_mode: pse
+    poe_type: type1-ieee802.3af
+  - name: GigabitEthernet1/0/5
+    type: 1000base-t
+    poe_mode: pse
+    poe_type: type1-ieee802.3af
+  - name: GigabitEthernet1/0/6
+    type: 1000base-t
+    poe_mode: pse
+    poe_type: type1-ieee802.3af
+  - name: GigabitEthernet1/0/7
+    type: 1000base-t
+    poe_mode: pse
+    poe_type: type1-ieee802.3af
+  - name: GigabitEthernet1/0/8
+    type: 1000base-t
+    poe_mode: pse
+    poe_type: type1-ieee802.3af
+  - name: GigabitEthernet1/0/9
+    type: 1000base-t
+    poe_mode: pse
+    poe_type: type1-ieee802.3af
+  - name: GigabitEthernet1/0/10
+    type: 1000base-t
+    poe_mode: pse
+    poe_type: type1-ieee802.3af
+  - name: GigabitEthernet1/0/11
+    type: 1000base-t
+    poe_mode: pse
+    poe_type: type1-ieee802.3af
+  - name: GigabitEthernet1/0/12
+    type: 1000base-t
+    poe_mode: pse
+    poe_type: type1-ieee802.3af
+  - name: GigabitEthernet1/0/13
+    type: 1000base-t
+    poe_mode: pse
+    poe_type: type1-ieee802.3af
+  - name: GigabitEthernet1/0/14
+    type: 1000base-t
+    poe_mode: pse
+    poe_type: type1-ieee802.3af
+  - name: GigabitEthernet1/0/15
+    type: 1000base-t
+    poe_mode: pse
+    poe_type: type1-ieee802.3af
+  - name: GigabitEthernet1/0/16
+    type: 1000base-t
+    poe_mode: pse
+    poe_type: type1-ieee802.3af
+  - name: GigabitEthernet1/0/17
+    type: 1000base-t
+    poe_mode: pse
+    poe_type: type1-ieee802.3af
+  - name: GigabitEthernet1/0/18
+    type: 1000base-t
+    poe_mode: pse
+    poe_type: type1-ieee802.3af
+  - name: GigabitEthernet1/0/19
+    type: 1000base-t
+    poe_mode: pse
+    poe_type: type1-ieee802.3af
+  - name: GigabitEthernet1/0/20
+    type: 1000base-t
+    poe_mode: pse
+    poe_type: type1-ieee802.3af
+  - name: GigabitEthernet1/0/21
+    type: 1000base-t
+    poe_mode: pse
+    poe_type: type1-ieee802.3af
+  - name: GigabitEthernet1/0/22
+    type: 1000base-t
+    poe_mode: pse
+    poe_type: type1-ieee802.3af
+  - name: GigabitEthernet1/0/23
+    type: 1000base-t
+    poe_mode: pse
+    poe_type: type1-ieee802.3af
+  - name: GigabitEthernet1/0/24
+    type: 1000base-t
+    poe_mode: pse
+    poe_type: type1-ieee802.3af
+  - name: TenGigabitEthernet1/0/1
+    type: 10gbase-x-x2
+  - name: TenGigabitEthernet1/0/2
```

Comment:
I think this one confuses me slightly. What delinates a "module" from an "interface"? I see a trillion examples of device definitions here where SFP's, SFP+'s. etc are designated as interfaces-- should those all be modules, instead?

It feels like that would make importing these device types very cumbersome if so, so I assume I'm probably misunderstanding.

Reply:
@wastrachan In this case, this is a X2 form factor with a fixed slot for uplink, so it would be non-modular.

Reply:
Got it, thanks! That's what had me stumped.

Reply:
![image](https://user-images.githubusercontent.com/8165027/231806746-d08df544-c34e-4c6e-a97e-bffbc8790a57.png)
Is this not the device? It appears that there are 24 fixed RJ-45 interfaces and 2 modular slots.

Reply:
@wastrachan so since there are 4 SFPs then, shouldn't we model this such that all 4 are in this file?

I have another question actually. If the X2 transceivers are NOT installed, does the devices configuration still show `TenGigabitEthernet1/0/1`? Or does it simply omit that?

Reply:
> @wastrachan so since there are 4 SFPs then, shouldn't we model this such that all 4 are in this file?

Missed this question. There are just two X2 optic bays, and no SFP's. The X2 optic can't be broken out into individual channels as far as I know. I think perhaps the ridiculous size of the X2 makes it's purpose a little misleading-- it's just an early, impractical version of a 10GBase-SR optic based on a Xenpack. It was pretty quickly superseded by the SFP+

![image](https://user-images.githubusercontent.com/1967731/231817287-5f6c248e-4755-4dce-a499-6f198bda0c30.png)

Reply:
> I have another question actually. If the X2 transceivers are NOT installed, does the devices configuration still show `TenGigabitEthernet1/0/1`? Or does it simply omit that

Most Cisco Catalyst devices will display all possible interfaces combinations for the hardware, I have only recently seen exceptions to this rule.

For this specific switch, you'll have two TenGigabit interfaces for the modules 10G X2 modules linked, and four Gigabit interfaces, 25 through 28, for an alternative adapter (CVR-X2-SFP).

IMO the system should still not have module bays in this case as both sets of interfaces are available from console.

Reply:
I agree, there should not be module bays, but I do still think that this definition file should have all interfaces then

Reply:
@wastrachan let me know if you agree, if so please add all 4 of the interfaces. thanks!

Reply:
I do agree, if the interfaces are statically defined in configuration they should be present in this definition. I'll update it shortly.